lib/manageiq/cross_repo/test_core.rb

```diff
@@ -0,0 +1,29 @@
+module ManageIQ::CrossRepo
+  class TestCore < TestBase
+    attr_reader :core_repo, :plugin_repos
+
+    def initialize(core_repo, plugin_repos)
+      @core_repo = Repository.new(core_repo)
+      @plugin_repos = plugin_repos.to_a.map { |repo| Repository.new(repo) }
+    end
+
+    def run
+      ensure_repo(core_repo)
+      plugin_repos.each { |plugin_repo| ensure_repo(plugin_repo) }
+
+      File.write(core_repo.path.join("bundler.d", "overrides.rb"),
+        plugin_repos.map { |plugin| "override_gem \"#{plugin.repo}\", :path => \"#{plugin.path}\"" }.join("\n")
+      ) unless plugin_repos.empty?
+
+      Dir.chdir(core_repo.path) do
+        require_relative core_repo.path.join("lib", "manageiq", "environment").to_s
+        ManageIQ::Environment.create_database_user if ENV["CI"]
```

Comment:
I feel like this should run something from the core repo instead of duplicating here (even though I know you tried to avoid changing core)... I wonder if we can run before_install directly inside the Bundler.with_clean_env block.

Reply:
> I wonder if we can run before_install directly inside the Bundler.with_clean_env block.

Like `system!("repos/ManageIQ/manageiq/tools/ci/before_install.sh")` ?